src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -105,7 +106,7 @@
  * check for 64 bit OS and 32 bit JVM is is64bit()
  *
  */
-public class Runtime extends Service implements MessageListener, RemoteMessageHandler, Gateway, LocaleProvider {
+public class Runtime extends Service implements MessageListener, ServiceLifeCyclePublisher, RemoteMessageHandler, Gateway, LocaleProvider {

```

Comment:
why is runtime the only thing that is publishing lifecycle stuff?  i mean. i'd expect services to implement a SerivceLifeCycle  interface (start/stop/created..etc..)    .. i guess services could publish when their lifecycle was called to notify other listeners, however that should happen after the service processes it's lifecycle request.

Reply:
Some events can be handled inside a class - started and stopped
Other events cannot be handled inside any service .. e.g. registered, released, created
The one thing other services are guaranteed is runtime exists - therefore since registered, released, and created are already handled by runtime, it was logical that it be responsible for broadcasting the remainder of the events started and stopped